applications/dot_marketplace-Phase3.md

```diff
@@ -0,0 +1,222 @@
+- Each milestone has its independent bidding system where multiple workers can place their bids
+- The publisher can select a bid as per the requirement and ratings of the worker and other criteria that can be added to a user account.
+- A worker can bid for multiple milestones of a single project based on their expertise.
+- A project reaches completion only if all milestones in the project are completed and approved by the publisher.
+- In our current implementation all milestones are independent, hence they can be completed and approved by the publisher irrespective of the overall project status.
+- Based on the requirements, a publisher can add more milestones to a project even after it is pushed to the market, provided the total number of milestones does not exceed 5 (configurable)
+- Decentralized IPFS Storage for project materials using NFTStorage Provider. Each material will have a unique CID that can be accessed by both Publisher and Worker.
+- Advance Search by task tags, ids & title.
+- The [decentralized court](https://github.com/WowLabz/dot-marketplace-v2) implemented in phase 2 is functional for each milestone of a project
+- All of the above features will be updated as a new feature for the existing marketplace pallet. Similarly, the selekatal UI will be updated to showcase the same.
+- A new file server written using the rocket framework will be provided for the integration with IPFS (using NftStorage). 
+
+The flow of tasking pallet with milestone based submission
+
+![Tasking-Court-Flow4 drawio](https://user-images.githubusercontent.com/66478092/190300655-1d2085b3-b728-4ced-8238-f262a9c5c5f8.png)
+
+
+### **Repository Hierarchy**
+
+```bash
+node
+├── build.rs
+├── Cargo.toml
+└── src
+    ├── chain_spec.rs
+    ├── cli.rs
+    ├── command.rs
+    ├── lib.rs
+    ├── main.rs
+    ├── rpc.rs
+    └── service.rs
+pallets
+├── pallet-chat
+│   ├── Cargo.toml
+│   ├── README.md
+│   └── src
+│       ├── benchmarking.rs
+│       ├── lib.rs
+│       ├── mock.rs
+│       └── tests.rs
+└── pallet-tasking
+    ├── Cargo.toml
+    ├── README.md
+    └── src
+        ├── benchmarking.rs
+        ├── lib.rs
+        ├── mock.rs
+        ├── utils.rs
+        └── tests.rs
+runtime
+├── build.rs
+├── Cargo.toml
+└── src
+    └── lib.rs
+```
+
+The current focus is to enhance the existing Substrate pallet and allied code base to get a basic yet functional marketplace up and running.
+
+### **Ecosystem Fit**
+
+We believe this work could be helpful for any Polkadot para-chains/ para-threads interested in including a marketplace with on-chain dispute resolution.
+
+- Almost all para-chains/ para-threads would find motivation in encouraging their community members to contribute meaningfully to their roadmap. This can be achieved by utilizing a marketplace like Dot Marketplace, where technical, marketing, or governance-centric projects can be published as bounties. And community members are invited to bid for and execute them.
```

Comment:
```suggestion
- Almost all parachains/parathreads would find motivation in encouraging their community members to contribute meaningfully to their roadmap. This can be achieved by utilizing a marketplace like Dot Marketplace, where technical, marketing, or governance-centric projects can be published as bounties. And community members are invited to bid for and execute them.
```